### ISSUE TITLE
Should we allow intermediat (non-100) weights?

### ISSUE THREAD

**Issue by thundernixon** [googlefonts/fontbakery#2420]
_I'm hoping that @davelab6, @m4rc1e, @laerm0, or others might be able to give their view this._

## Observed behaviour

[Fira Code](https://github.com/tonsky/FiraCode/tree/d33ca1d82964d663a3b8b453d3769fd37ee3455b) currently has a named instance, "Retina," halfway between Regular and Medium, to allow users a bit more of a choice on font weight in their text editors.

To me, this seems like it should (ideally) be a valid design choice – it makes sense to offer users a bit more choice in font weight, without having to do something more complex like adding in a "grade" axis.

For instance, IBM Plex has a "Text" instance with a weight value of `450` (https://github.com/IBM/plex/blob/master/scripts/data/weights.js).

I know that the GF Docs say that [weights must be in a valid CSS value of 100–900](https://github.com/googlefonts/gf-docs/tree/master/METADATA#fontweight). But, might this be something that becomes less rigid with variable fonts?

Currently, this causes a `fail` in FontBakery:

<details>
<summary>🔥 <b>FAIL:</b> Variable font weight coordinates must be multiples of 100.</summary>

* [com.google.fonts/check/varfont_weight_instances](https://github.com/googlefonts/fontbakery/search?q=com.google.fonts/check/varfont_weight_instances)
* 🔥 **FAIL** Found an variable font instance with 'wght'=450.0. This should instead be a multiple of 100.

</details>

It also has caused some problems in FontMake, as the Fira Code source currently labels the Retina instance with `weightClass: 900`, presumably just to use an open slot.

For now, I've tried:
- A test of giving both "Regular" and "Retina" `400` in GlyphsApp results in "Regular" having a value of `364.70589` (obviously not great). [FiraCode-VF-retina.ttf.zip](https://github.com/googlefonts/fontbakery/files/3009378/FiraCode-VF-retina.ttf.zip)

- unchecking this as an `active` instance in GlyphsApp, which prevents it from building into static or variable fonts. This makes things work at a basic level, but misses part of the design. [FiraCode-VF.ttf.zip](https://github.com/googlefonts/fontbakery/files/3009379/FiraCode-VF.ttf.zip)

## Expected behaviour?

Should we allow intermediate weights, or just put a blanket "no" on it to keep things easy for us (though not necessarily as good as possible for font users)? Do VFs change this at all? 

If we can allow intermediate weights, what might be a good way to do so? Should I just give the "Retina" instance a "Medium," `500` weight class, then bump up the existing Medium to `600`? Or is there a better way I'm not seeing?

## Resources and exact process needed to replicate

You should be able to replicate this issue if you build then check from the GlyphsApp source at [2eb3fc6b](https://github.com/thundernixon/FiraCode/blob/2eb3fc6b9ee8f8c06ac8ed31c95bc552b631531f/FiraCode.glyphs)

**Comment by laerm0** [comment]
I can't see any logical reason not to allow weights not in multiples of 100, but I am not a web developer so I dunno how CSS would handle it.

**Comment by thundernixon** [comment]
As I understand it, you can’t get a non-100-multiple weight from CSS today, if you’re calling in a static font. However, you *could* set that “Retina” static as a weight value like 500 when pulling it in as a static font via @font-face, and then call it as 500.

In a variable font, so long as the @font-face calls a VF file and sets a font weight range, the user could call any font weight value in that range.

There is also a spec to allow CSS users to call named instances in a font:
https://www.w3.org/TR/css-fonts-4/#font-named-instance

**Comment by thundernixon** [comment]
As a small update here, I've found a way to set up the instance in GlyhpsApp such that it exports a font as expected: "Retina" gets a custom parameter of `weightClass: 450`, and a glyphs menu-weight of "Normal." Tested in axis-praxis, it seems to work well ([FiraCode-VF.ttf.zip](https://github.com/googlefonts/fontbakery/files/3010086/FiraCode-VF.ttf.zip)).

In this case, is there still a reason to avoid non-100-multiples, or is this check overly-strict?

---

Instance setup in Glyphs:

![image](https://user-images.githubusercontent.com/7355414/55029266-0fed7680-4fe0-11e9-94fa-20d31ee1ab06.png)

![image](https://user-images.githubusercontent.com/7355414/55029310-2b588180-4fe0-11e9-9375-b9f808223435.png)

**Comment by TBBle** [comment]
> I know that the GF Docs say that [weights must be in a valid CSS value of 100–900](https://github.com/googlefonts/gf-docs/tree/master/METADATA#fontweight). But, might this be something that becomes less rigid with variable fonts?

Poking around, I noticed that the [CSS Fonts 4 current draft](https://www.w3.org/TR/css-fonts-4/#font-weight-prop) removes the multiple-of-100 requirement from `font-weight`, as part of allowing variable fonts to function correctly on the Web. So that indeed turns out to be becoming less rigid, at least in CSS. Hopefully GF rules will follow suit as that moves through the ecosystem.

**Comment by davelab6** [comment]
This check is in the gf profile (com.google.fonts/check/varfont_weight_instances) not universal, and gf continues to require this due to workspace code. @m4rc1e did this intersect with your latest name table check rewrite?

**Comment by m4rc1e** [comment]
Yes it did intersect with my new checks so I removed it. If a user is checking a font using the google fonts profile and their instances are not multiples of 100, it will still error on the new check.
